src/openrct2/actions/RideEntranceExitRemoveAction.hpp

```diff
@@ -0,0 +1,194 @@
+ * Copyright (c) 2014-2019 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../ride/Ride.h"
+#include "../ride/Station.h"
+#include "../world/Entrance.h"
+#include "GameAction.h"
+
+DEFINE_GAME_ACTION(RideEntranceExitRemoveAction, GAME_COMMAND_REMOVE_RIDE_ENTRANCE_OR_EXIT, GameActionResult)
+{
+private:
+    CoordsXY _loc;
+    NetworkRideId_t _rideIndex;
+    uint8_t _stationNum;
+    bool _isExit;
+
+public:
+    RideEntranceExitRemoveAction() = default;
+
+    RideEntranceExitRemoveAction(CoordsXY loc, ride_id_t rideIndex, uint8_t stationNum, bool isExit)
+        : _loc(loc)
+        , _rideIndex(rideIndex)
+        , _stationNum(stationNum)
+        , _isExit(isExit)
+    {
+    }
+
+    uint16_t GetActionFlags() const override
+    {
+        return GameAction::GetActionFlags();
+    }
+
+    void Serialise(DataSerialiser & stream) override
+    {
+        GameAction::Serialise(stream);
+
+        stream << DS_TAG(_loc) << DS_TAG(_rideIndex) << DS_TAG(_stationNum) << DS_TAG(_isExit);
+    }
+
+    GameActionResult::Ptr Query() const override
+    {
+        if (_rideIndex >= MAX_RIDES || _rideIndex == RIDE_ID_NULL)
+        {
+            log_warning("Invalid game command for ride %u", uint32_t(_rideIndex));
+            return std::make_unique<GameActionResult>(GA_ERROR::INVALID_PARAMETERS, STR_NONE);
+        }
+
+        Ride* ride = get_ride(_rideIndex);
+        if (ride == nullptr || ride->type == RIDE_TYPE_NULL)
+        {
+            log_warning("Invalid ride id %u for entrance/exit removal", _rideIndex);
+            return std::make_unique<GameActionResult>(GA_ERROR::INVALID_PARAMETERS, STR_NONE);
+        }
+
+        if (ride->status != RIDE_STATUS_CLOSED)
+        {
+            return MakeResult(GA_ERROR::INVALID_PARAMETERS, STR_MUST_BE_CLOSED_FIRST);
+        }
+
+        if (ride->lifecycle_flags & RIDE_LIFECYCLE_INDESTRUCTIBLE_TRACK)
+        {
+            return MakeResult(GA_ERROR::INVALID_PARAMETERS, STR_NOT_ALLOWED_TO_MODIFY_STATION);
+        }
+
+        bool found = false;
+        TileElement* tileElement = map_get_first_element_at(_loc.x / 32, _loc.y / 32);
+
+        do
+        {
+            if (tileElement == nullptr)
+                break;
+
+            if (tileElement->GetType() != TILE_ELEMENT_TYPE_ENTRANCE)
+                continue;
+
+            if (tile_element_get_ride_index(tileElement) != _rideIndex)
+                continue;
+
+            if (tileElement->AsEntrance()->GetStationIndex() != _stationNum)
+                continue;
+
+            if ((GetFlags() & GAME_COMMAND_FLAG_5) && !(tileElement->flags & TILE_ELEMENT_FLAG_GHOST))
```

Comment:
Is this flag actually correct?

Reply:
Yeah it would make more sense if it was GAME_COMMAND_FLAG_GHOST. I think I'll change it.

Reply:
Strangely track designs actually have their own version of this whole game command instead of nesting them.